suse/sle-hpc-15.x/sle-hpc-15-sp4/install_utils.sh

```diff
@@ -0,0 +1,135 @@
+
+# import package signing keys
+rpm --import https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+# add repository
+zypper -n addrepo -f -g https://yum.repos.intel.com/oneapi oneAPI
+# fetch key
+zypper --non-interactive --gpg-auto-import-keys refresh
+
+# list all packages
+# sudo -E zypper pa -ir oneAPI
+#-------------------------------------------------------------------
+
+#-------------------------------------------------------------------
+# Nvidia provide certified packages for SLES 15, so we only need to add the repositories and install the packages
+#-------------------------------------------------------------------
+# import cuda signing key
+rpm --import https://developer.download.nvidia.com/compute/cuda/repos/sles15/x86_64/D42D0685.pub
+
+# CUDA driver (nvidia provides a repo file)
+zypper addrepo -f -g https://developer.download.nvidia.com/compute/cuda/repos/sles15/x86_64/cuda-sles15.repo
+# fetch key
+zypper --non-interactive --gpg-auto-import-keys refresh
+
+#-------------------------------------------------------------------
+# Container Repository
+#-------------------------------------------------------------------
+# Docker is shipped with SLES by default
+# with SLES HPC we need to enable the Container repository
+SUSEConnect -p sle-module-containers/15.4/x86_64
+
+#-------------------------------------------------------------------
+# nvidia container repo
+#-------------------------------------------------------------------
+# see https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html
+# Check https://nvidia.github.io/libnvidia-container
+zypper addrepo -f -g https://nvidia.github.io/libnvidia-container/sles15.3/libnvidia-container.repo
+# fetch key
+zypper --non-interactive --gpg-auto-import-keys refresh
+
+#-------------------------------------------------------------------
+# Add SUSE Package Hub
+# byacc is only in packagehub
+SUSEConnect -p PackageHub/15.4/x86_64
+#-------------------------------------------------------------------
+
+#
+## SLES HPC ship with many HPC packages already, so no need to build it - simple install is enough
+#
+# as the default cloud image does only install parts of the HPC toolchain we use the pattern
+zypper install -y --type pattern hpc_libraries
+
+## Lmod is an advanced environment module system that allows the installation of multiple versions of a program or shared library, and helps configure the system environment for the use of a specific version.
+## the modulefile path is /usr/share/lmod/modulefiles
+zypper install lua-lmod
+source /usr/share/lmod/lmod/init/bash
+
+#
+zypper install -y \
+    numactl \
+    byacc \
+    gtk2 \
+    atk \
+    m4 \
+    tcsh \
+    gcc-fortran \
+    elfutils \
+    kernel-macros \
+    binutils \
+    kernel-headers \
+    kernel-source \
```

Comment:
Is it a good idea to hold kernel from updating as most of the hpc packages are dependent on kernel?

Reply:
We do not change the kernel and the kernel api within a ServicePack. This is one of the differences and the promise of a Enterprise Linux distribution vs. community distributions - a stable kABI.